test/dates/ranges.jl

```diff
@@ -482,10 +482,10 @@ function testmonthranges2(n)
 end
 testmonthranges2(100000)
 
-@test length(Dates.Year(1):Dates.Year(10)) == 10
+@test length(Dates.Year(1):Dates.Year(1):Dates.Year(10)) == 10
```

Comment:
I think it's dangerous to define ranges with unitful quantities if you don't explicitly specify the step, because with unitful quantities we tend to think that there's an underlying physical reality independent of the units you choose to express the quantities in. Why should `1m:10m` give a range of length 10 whereas `1000mm:10000mm` gives you a range of length 9001? Both of those ranges are defined with the same endpoints.

In my mind the only reasonable approach is to force the user to specify the step or the desired length of the range.